index_documents_test.go

```diff
@@ -1383,6 +1383,129 @@ func TestIndex_DeleteDocuments(t *testing.T) {
 	}
 }
 
+func TestIndex_DeleteDocumentsByFilter(t *testing.T) {
+	type args struct {
+		UID            string
+		client         *Client
+		filterToDelete interface{}
+		filterToApply  []string
+		documentsPtr   []docTestBooks
+	}
+	tests := []struct {
+		name     string
+		args     args
+		wantResp *TaskInfo
+	}{
+		{
+			name: "TestIndexDeleteDocumentsByFilterString",
+			args: args{
+				UID:            "1",
+				client:         defaultClient,
+				filterToApply:  []string{"book_id"},
+				filterToDelete: "book_id = 123",
+				documentsPtr: []docTestBooks{
```

Comment:
What does ptr mean?

Reply:
These are auto-generated tests, I think it's for pointer.